### PR TITLE
docs: updated docstrings using ruff's autofix feature

### PR DESCRIPTION
**Pull request by anushkumarv**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
Updates the docstrings of several functions by Ruff's guidelines 
[issue 1698](https://github.com/OpenDevin/OpenDevin/issues/1698)

**Give a brief summary of what the PR does, explaining any non-trivial design decisions**
It updates docstrings using Ruff's autofix feature.
The changes were made by adding the below to `pyproject.toml` file and running `ruff check . --fix`

```
[tool.ruff.lint]
select = ["D"]
# ignore warnings for missing docstrings
ignore = ["D1"]

[tool.ruff.lint.pydocstyle]
convention = "google"
```

**Additional details**
I was unable to commit the above changes to `pyprject.toml` unless I moved the section `[tool.poetry.group.evaluation.dependencies]` to the end of the file. 
The pre-commit check `pyproject-fmt` seems always add a new line after the section `[tool.poetry.group.evaluation.dependencies]` (even if I tried a 100 times). The only way I was able to pass the check was to move the section to the end of file. Alternatively, updating `pyproject-fmt` from `1.7.0` to `2.0.4` seemed to go through without issues

Please let me know if the changes to `pyprject.toml` should go through or not because as of now, ruff's pre-commit check does not include docstring validations.

### CODE REVIEW

**Comment by tobitege** [comment]
Thanks for working on this!
The new setting is not applied to the `dev_config/python/.pre-commit-config.yaml` file, though?
https://github.com/OpenDevin/OpenDevin/blob/46edb4b15bfd0db17a352ac324401021c06fa80f/dev_config/python/.pre-commit-config.yaml
Ideally, this should run on any newly committed files, too. 🤔

**Comment by anushkumarv** [comment]
Yes for the settings to work, I believe the changes also need to be add to this file.
https://github.com/OpenDevin/OpenDevin/blob/46edb4b15bfd0db17a352ac324401021c06fa80f/dev_config/python/.pre-commit-config.yaml

Should I include them in the PR?

**Comment by tobitege** [comment]
> Yes for the settings to work for newly commited files, I believe the changes also need to be add to this file. https://github.com/OpenDevin/OpenDevin/blob/main/dev_config/python/ruff.toml
> 
> Should I include them in the PR?

Hello @neubig,
would you prefer to have this in the same PR or maybe split up (the latter is for "dev")?

**Comment by tobitege** [comment]
Hey @anushkumarv , just need to fix the interim 2 conflicts and then this should be good to go.

File: opendevin/runtime/utils/source.py
Comment (neubig) [inline opendevin/runtime/utils/source.py:1]: This is not just docstring changes, could you take a look? Once this is fixed I think we're ready to merge!

Comment (anushkumarv) [reply opendevin/runtime/utils/source.py:1]: Thats my bad. I did a rebase from main onto my branch earlier this morning and looks like I made a mistake with this file. I deleted the file to keep it insync with main

**Review by neubig — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**